rust/core-lib/src/file.rs

```diff
@@ -110,9 +110,15 @@ impl FileManager {
     pub fn open(&mut self, path: &Path, id: BufferId)
         -> Result<Rope, FileError>
     {
-        let (rope, info) = try_load_file(path)?;
+        let (rope, info) = if path.exists() {
```

Comment:
This is a race if the file is deleted, but maybe it's ok for that to be an error.

Reply:
Yea that sounds okay in this case, but there are definitely some other bits of possible strangeness here; see my top level comment.